### DISCUSSION TITLE
Signals, Angular, questions and hopefully answers

### DISCUSSION THREAD

**Discussion by SanderElias** [SanderElias/Samples#48] [Ideas]
In this discussion there is room to talk about 🚦 and all that comes too it.

**Comment by leomayer** [comment]
Some questions arise about the Signal especially it sounds brand new. Is there somewhere a summary link with the different functions, like effect, signal, mutate, etc...? The other q which is quite unclear if a ChangeDetection runs automatically after a change was applied? Or is there a way to collect signals?
I'm thinking on a Tree update of several nodes, but the tree itself remains unchanged but e.g. would enable additional chip-tags which cause different heights. So, an update would be great to have only once instead of each tree nodeand chip

**Reply by SanderElias** [reply]
Signals aren't a "new" concept, but they are new to Angular. 
Angular came up with its [own implementation](https://github.com/angular/angular/tree/main/packages/core/src/signals). On that link, you can find the readme with all the details. 
Be aware that all of this is not even in "developer preview", so you should not be using it at all. There will be an RFC first, and then it will probably go into "developer preview". (that is the time you can start using it, but carefully)

The whole point of adding signals is to get rid of change detection completely. (in a faraway distant future!).
That means if some data is updated, only the bindings showing this data will be updated automaticly

**Comment by runnyren** [comment]
In Angular, ChangeDetection happens automatically after changes are applied to the data model. Angular uses Zone.js to detect changes and fire ChangeDetection. When an event occurs in Angular, such as a mouse click or a data change, Zone.js fires a ChangeDetection to detect changes in the data model and update the user [interface](https://www.61saat.com/korayspor-spor-ceketlerinde-buyuk-firsatlar).

**Reply by SanderElias** [reply]
Is there a question in there I missed somehow?
